File: src/test/resources/oracle/commits/rm-action-test-8187d840c38c0a1eaf8617c1290f2782084dd059/GuestAccount.java
public class GuestAccount {
    protected String username;

    public GuestAccount(String username) {
        this.username = username;
    }

    public String displayName() {
        return username.trim().toUpperCase();
    }

    public boolean canBrowseCatalog() {
        return username.length() > 0;
    }
}

File: src/test/resources/oracle/commits/rm-action-test-0a5085070f80f44c03645e1557c2bba8da3b335e/typescript/GuestAccount.ts
export class GuestAccount {
    protected username: string;

    constructor(username: string) {
        this.username = username;
    }

    displayName(): string {
        return this.username.trim().toUpperCase();
    }

    canBrowseCatalog(): boolean {
        return this.username.length > 0;
    }
}
